spinn_front_end_common/utilities/helpful_functions.py

```diff
@@ -303,6 +304,19 @@ def convert_vertices_to_core_subset(vertices, placements):
     return core_subsets
 
 
+def find_executable_start_type(machine_vertex, graph_mapper=None):
+    has_binary = isinstance(machine_vertex, AbstractHasAssociatedBinary)
+
+    if not has_binary:
+        return None
+    elif graph_mapper is not None:
```

Comment:
else not needed...

Reply:
else definately needed. machine graph levels would return None if that else wasnt there.

Reply:
You misunderstand - the above is a "return", so "elif" can become just "if".  Not critical though...

Reply:
fixed